components/server/src/workspace/workspace-starter.ts

```diff
@@ -951,17 +960,29 @@ export class WorkspaceStarter {
 
             const snapshot = new SnapshotInitializer();
             snapshot.setSnapshot(context.snapshotBucketId);
-            const { git } = await this.createGitInitializer(traceCtx, workspace, context, user);
+            const { initializer } = await this.createCommitInitializer(traceCtx, workspace, context, user);
             const init = new PrebuildInitializer();
             init.setPrebuild(snapshot);
-            init.setGit(git);
+            if (initializer instanceof CompositeInitializer) {
```

Comment:
Using this API seems noisy... is there a reason that not we have non-composite initializers? And if there is: Could we encoded that with a boolean flag, and get rid of these `instanceof` branches by making all initializers composites, and just tick that flag?

Reply:
Yes, it is noisy. We need the compositeinitializer for non-prebuilds. So in case of prebuilds we decompose the structure and translate to the prebuild initializer shape.